packages/suite/src/utils/wallet/cardanoUtils.ts

```diff
@@ -0,0 +1,262 @@
+import {
+    CARDANO,
+    CardanoAddressType,
+    CardanoCertificate,
+    CardanoCertificateType,
+    CardanoOutput,
+} from 'trezor-connect';
+import type { types } from '@fivebinaries/coin-selection';
+import { amountToSatoshi, formatAmount, formatNetworkAmount } from '@wallet-utils/accountUtils';
+import { Account } from '@wallet-types';
+import {
+    Output,
+    PrecomposedTransactionFinal,
+    PrecomposedTransactionFinalCardano,
+} from '@wallet-types/sendForm';
+import BigNumber from 'bignumber.js';
+import { PoolsResponse, StakePool } from '@suite/types/wallet/cardanoStaking';
+import { CARDANO_DEFAULT_TTL } from '@wallet-constants/sendForm';
+
+export const loadCardanoLib = async () => {
+    const lib = await import(
+        /* webpackChunkName: "@fivebinaries/coin-selection" */ '@fivebinaries/coin-selection'
+    );
+    return lib;
+};
+
+export const getProtocolMagic = (accountSymbol: Account['symbol']) =>
+    accountSymbol === 'ada' ? CARDANO.PROTOCOL_MAGICS.mainnet : 1097911063;
+
+export const getNetworkId = (accountSymbol: Account['symbol']) =>
+    accountSymbol === 'ada' ? CARDANO.NETWORK_IDS.mainnet : CARDANO.NETWORK_IDS.testnet;
+
+export const getAddressType = (accountType: Account['accountType']) =>
+    accountType === 'normal' ? CardanoAddressType.BASE : CardanoAddressType.BYRON;
+
+export const getStakingPath = (
+    accountType: Account['accountType'],
+    accountIndex: Account['index'],
+) => `m/${accountType === 'normal' ? 1852 : 44}'/1815'/${accountIndex}'/2/0`;
+
+export const getChangeAddressParameters = (account: Account) => {
+    if (!account.addresses || account.networkType !== 'cardano') return;
+    const stakingPath = getStakingPath(account.accountType, account.index);
+    // Find first unused change address or fallback to the last address if all are used (should not happen)
+    const changeAddress =
+        account.addresses.change.find(a => !a.transfers) ||
+        account.addresses.change[account.addresses.change.length - 1];
+
+    return {
+        address: changeAddress.address,
+        addressParameters: {
+            path: changeAddress.path,
+            addressType: getAddressType(account.accountType),
+            stakingPath,
+        },
+    };
+};
+
+export const transformUserOutputs = (
+    outputs: Output[],
+    accountTokens: Account['tokens'],
+    maxOutputIndex?: number,
+): types.UserOutput[] =>
+    outputs.map((output, i) => {
+        const setMax = i === maxOutputIndex;
+        const amount = output.amount === '' ? undefined : amountToSatoshi(output.amount, 6);
```

Comment:
would be great if we could use number of decimals from networks.ts.Could it be feasible here? Just brainstorming at the moment.